projects/driver_controls/test/test_pedal_calibration.c

```diff
@@ -0,0 +1,71 @@
+#include "ads1015_def.h"
+#include "debouncer.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "input_event.h"
+#include "interrupt.h"
+#include "pedal_calibration.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+static PedalCalibrationStorage s_storage;
+static Ads1015Storage s_ads1015_storage;
+static ThrottleCalibrationData s_throttle_calibration_data;
+static DebouncerInfo s_debouncer_info;
+static bool s_continue_flag;
+#define TEST_PEDAL_CALIBRATION_ADC_CHANNEL_A ADS1015_CHANNEL_0
+#define TEST_PEDAL_CALIBRATION_ADC_CHANNEL_B ADS1015_CHANNEL_1
+#define TEST_PEDAL_CALIBRATION_BRAKE_PERCENTAGE 30
+#define TEST_PEDAL_CALIBRATION_COAST_PERCENTAGE 10
+
+static void prv_wait_for_button(void) {
+  s_continue_flag = false;
+  while (!s_continue_flag) {
+  }
+}
+
+static void prv_continue_calibration(const GPIOAddress *address, void *context) {
+  s_continue_flag = true;
```

Comment:
Is there a reason you don't just use the context parameter? Also, you'll need to make it your flag volatile or the compiler will probably optimize it out.

Reply:
I will pass it as context!